server/controllers/finance/cash.js

```diff
@@ -177,6 +177,8 @@ function listPayment(options) {
 
   // @TODO Support ordering query (reference support for limit)?
   filters.setOrder('ORDER BY cash.date DESC');
+ 
+  filters.custom('invoice_uuid', 'cash.uuid IN (SELECT cash_item.cash_uuid FROM cash_item WHERE cash_item.invoice_uuid = HUID(?))');
```

Comment:
:+1:  I haven't validated this for correctness, but it's a good re-use of tools.